### DISCUSSION TITLE
New Sprite type supporting alpha channel

### DISCUSSION THREAD

**Discussion by Tarzan1163** [Bodmer/TFT_eSPI#2286] [New processor/display support or capability request]
**Problem**
In my project a have sprite that holds some antialiased graphic. Part of this graphic is repainted with transparent color (masked). However when i push this sprite to another sprite, then part with alpha is already premultiplied and thus antialiased parts are useles.
I cannot paint directly into destination sprite because of that masking step.

**Use case**
1) create Sprite with transparent background
2) paint some antialiased graphic with opaque color (e.g. fillSmoothCircle())
3) use pushSprite() or pushToSprite()
4) see properly blended graphics (antialiased parts)

**Solution**
Introduce sprite that holds alpha and blending with background color is done only when pushSprite() or pustToSprite() is called.

Would it be possible to add such sprite type into library?

**Comment by Bodmer** [comment]
Adding an alpha channel to sprites is on the TO DO list but it will be some time before it gets added. This capability would then allow masking/transparency/opacity features.

**Reply by MR-XieXuan** [reply]
Expect

**Comment by DoctorU** [comment]
I've been thinking on this.

You could create two sprites, one for your image, and a grayscale one, from alpha channel, for your mask. Then interpolate.

I'll give it some more thought and try to create a demo.

**Reply by uzi18** [reply]
@Tarzan1163 any news?

**Reply by DoctorU** [reply]
I'm afraid not. Too little free time at the moment.
